### ISSUE TITLE
dbt run: Invalid input syntax: Use `DROP TABLE` to drop a table.

### ISSUE THREAD

**Issue by maingoh** [risingwavelabs/dbt-risingwave#56]
When running `dbt run --model foo --debug` with `dbt-risingwave=1.8.1`, I get the following error. It looks related to https://github.com/risingwavelabs/dbt-risingwave/pull/51 as it is working with `dbt-risingwave==1.8.0`. 
We can see a suspicious `10:29:57  Postgres adapter: Postgres error: Failed to run the query` (shouldn't it use the risingwave adapter ?)

```
10:29:56  Sending event: {'category': 'dbt', 'action': 'invocation', 'label': 'start', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa98e64c0>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa9e006a0>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aaae3c880>]}
10:29:56  Running with dbt=1.8.1
10:29:56  running dbt with arguments {'printer_width': '80', 'indirect_selection': 'eager', 'write_json': 'True', 'log_cache_events': 'False', 'partial_parse': 'True', 'cache_selected_only': 'False', 'profiles_dir': '/tmp/example/.dbt', 'version_check': 'True', 'debug': 'True', 'log_path': '/tmp/example/dbt/logs', 'fail_fast': 'False', 'warn_error': 'None', 'use_colors': 'True', 'use_experimental_parser': 'False', 'no_print': 'None', 'quiet': 'False', 'warn_error_options': 'WarnErrorOptions(include=[], exclude=[])', 'introspect': 'True', 'invocation_command': 'dbt run --model foo --debug', 'static_parser': 'True', 'target_path': 'None', 'log_format': 'default', 'send_anonymous_usage_stats': 'True'}
10:29:56  Sending event: {'category': 'dbt', 'action': 'project_id', 'label': '04c4bfac-3c0d-4947-b86e-ae1ec4489263', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa634df40>]}
10:29:56  Sending event: {'category': 'dbt', 'action': 'adapter_info', 'label': '04c4bfac-3c0d-4947-b86e-ae1ec4489263', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa6218970>]}
10:29:56  Registered adapter: risingwave=1.8.1
10:29:56  checksum: ebc509fb14e151ea268e89b60fd5abf912ed99129f87516462718fa2f14f4838, vars: {}, profile: , target: , version: 1.8.1
10:29:56  Partial parsing enabled: 0 files deleted, 0 files added, 0 files changed.
10:29:56  Partial parsing enabled, no changes found, skipping parsing
10:29:56  Sending event: {'category': 'dbt', 'action': 'load_project', 'label': '04c4bfac-3c0d-4947-b86e-ae1ec4489263', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa601a340>]}
10:29:57  Sending event: {'category': 'dbt', 'action': 'resource_counts', 'label': '04c4bfac-3c0d-4947-b86e-ae1ec4489263', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa5f97940>]}
10:29:57  Found 17 models, 451 macros
10:29:57  Sending event: {'category': 'dbt', 'action': 'runnable_timing', 'label': '04c4bfac-3c0d-4947-b86e-ae1ec4489263', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa5fc4f40>]}
10:29:57  
10:29:57  Acquiring new risingwave connection 'master'
10:29:57  Acquiring new risingwave connection 'list_dev'
10:29:57  Using risingwave connection "list_dev"
10:29:57  On list_dev: /* {"app": "dbt", "dbt_version": "1.8.1", "profile_name": "example", "target_name": "dev", "connection_name": "list_dev"} */

    select distinct nspname from pg_namespace
  
10:29:57  Opening a new connection, currently in state init
10:29:57  SQL status: SELECT 4 in 0.085 seconds
10:29:57  On list_dev: Close
10:29:57  Re-using an available connection from the pool (formerly list_dev, now list_dev_public)
10:29:57  Using risingwave connection "list_dev_public"
10:29:57  On list_dev_public: /* {"app": "dbt", "dbt_version": "1.8.1", "profile_name": "example", "target_name": "dev", "connection_name": "list_dev_public"} */
select
    'dev' as database,
    rw_relations.name as name,
    rw_schemas.name as schema,
    CASE WHEN relation_type = 'materialized view' THEN
      'materialized_view'
      else relation_type
    END AS type
    from rw_relations join rw_schemas on schema_id=rw_schemas.id
    where rw_schemas.name not in ('rw_catalog', 'information_schema', 'pg_catalog')
    and relation_type in ('table', 'view', 'source', 'sink', 'materialized view', 'index')
    AND rw_schemas.name = 'public'
  
10:29:57  Opening a new connection, currently in state closed
10:29:57  SQL status: SELECT 29 in 0.089 seconds
10:29:57  On list_dev_public: Close
10:29:57  Sending event: {'category': 'dbt', 'action': 'runnable_timing', 'label': '04c4bfac-3c0d-4947-b86e-ae1ec4489263', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa5f6c9a0>]}
10:29:57  Concurrency: 1 threads (target='dev')
10:29:57  
10:29:57  Began running node model.example.foo
10:29:57  1 of 1 START sql table_with_connector model public.foo ............ [RUN]
10:29:57  Re-using an available connection from the pool (formerly list_dev_public, now model.example.foo)
10:29:57  Began compiling node model.example.foo
10:29:57  Writing injected SQL for node "model.example.foo"
10:29:57  Began executing node model.example.foo
10:29:57  Determining configuration changes on: "dev"."public"."foo"
10:29:57  Using risingwave connection "model.example.foo"
10:29:57  On model.example.foo: /* {"app": "dbt", "dbt_version": "1.8.1", "profile_name": "example", "target_name": "dev", "node_id": "model.example.foo"} */

    
    with index_info as (
    select
        i.relname                                   as name,
        'btree'                                     as method,
        ix.indisunique                              as "unique",
        a.attname                                   as attname,
        array_position(ix.indkey, a.attnum)        as ord
    from pg_index ix
    join pg_class i
        on i.oid = ix.indexrelid
    join pg_class t
        on t.oid = ix.indrelid
    join pg_namespace n
        on n.oid = t.relnamespace
    join pg_attribute a
        on a.attrelid = t.oid
        and a.attnum = ANY(ix.indkey)
    where t.relname = 'foo'
      and n.nspname = 'public'
      and t.relkind in ('r', 'm')
    )
    select name, method, "unique", array_to_string(array_agg(attname order by ord), ',') as column_names from index_info
    group by 1, 2, 3
    order by 1, 2, 3;

  
10:29:57  Opening a new connection, currently in state closed
10:29:57  SQL status: SELECT 1 in 0.110 seconds
10:29:57  Applying UPDATE INDEXES to: "dev"."public"."foo"
10:29:57  Writing runtime sql for node "model.example.foo"
10:29:57  Using risingwave connection "model.example.foo"
10:29:57  On model.example.foo: /* {"app": "dbt", "dbt_version": "1.8.1", "profile_name": "example", "target_name": "dev", "node_id": "model.example.foo"} */
drop index if exists "public"."foo";
10:29:57  Postgres adapter: Postgres error: Failed to run the query

Caused by:
  Invalid input syntax: Use `DROP TABLE` to drop a table.


10:29:57  On model.example.foo: Close
10:29:57  Database Error in model foo (models/debezium_kafka_sources/foo.sql)
  Failed to run the query
  
  Caused by:
    Invalid input syntax: Use `DROP TABLE` to drop a table.
  compiled code at target/run/example/models/debezium_kafka_sources/foo.sql
10:29:57  Sending event: {'category': 'dbt', 'action': 'run_model', 'label': '04c4bfac-3c0d-4947-b86e-ae1ec4489263', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa539cac0>]}
10:29:57  1 of 1 ERROR creating sql table_with_connector model public.foo ... [ERROR in 0.20s]
10:29:57  Finished running node model.example.foo
10:29:57  Connection 'master' was properly closed.
10:29:57  Connection 'model.example.foo' was properly closed.
10:29:57  
10:29:57  Finished running 1 table with connector model in 0 hours 0 minutes and 0.47 seconds (0.47s).
10:29:57  Command end result
10:29:57  
10:29:57  Completed with 1 error and 0 warnings:
10:29:57  
10:29:57    Database Error in model foo (models/debezium_kafka_sources/foo.sql)
  Failed to run the query
  
  Caused by:
    Invalid input syntax: Use `DROP TABLE` to drop a table.
  compiled code at target/run/example/models/debezium_kafka_sources/foo.sql
10:29:57  
10:29:57  Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
10:29:57  Resource report: {"command_name": "run", "command_wall_clock_time": 1.1540309, "process_user_time": 2.052646, "process_kernel_time": 0.092028, "process_mem_max_rss": "104236", "process_out_blocks": "2048", "command_success": false, "process_in_blocks": "0"}
10:29:57  Command `dbt run` failed at 11:29:57.525881 after 1.16 seconds
10:29:57  Sending event: {'category': 'dbt', 'action': 'invocation', 'label': 'end', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa98e64c0>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa4317a00>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x7f1aa5fd0ee0>]}
10:29:57  Flushing usage events

```

**Comment by MattiasMTS** [comment]
Can you provide the sql query you are trying this out on? I'm not able to reproduce this with the following:

```sql
-- models/foo.sql
{{
    config(
        materialized="table",
        indexes=[
            {"columns": ["t"]},
        ],
    )
}}
select 1 as t
```
running  ` dbt run -s foo --full-refresh --debug` works, the ddl:

```
  create table if not exists "dev"."dev_mattiassjodin"."foo"


  as
select 1 as t
  ;
23:02:17  SQL status: INSERT 0 1 in 1.291 seconds
23:02:17  Using risingwave connection "model.core.foo"
23:02:17  On model.core.foo: /* {"app": "dbt", "dbt_version": "1.8.9", "profile_name": "dev", "target_name": "dev", "node_id": "model.core.foo"} */

    create index if not exists
  "__dbt_index_foo_t"
  on "dev"."dev_mattiassjodin"."foo"
  (t);
```

As for the postgres errors, this is expected given this adapter is inheriting most of the feature from the dbt-postgres adapter (as e.g. dbt-redshift is doing too). The changes in #51 is only the `materialized=table_with_connector` and not `materialized=table`, so I don't think that is the cause of the problem.

**Comment by MattiasMTS** [comment]
Fix introduced https://github.com/risingwavelabs/dbt-risingwave/pull/57 @maingoh . Can you point to that branch and give it a try?

**Comment by maingoh** [comment]
It works with the fix from #57 ! Thank you :)

Is there a release planned ?

**Comment by chenzl25** [comment]
It will be shipped with release v1.9.
